File: frontend/src/types/health.ts
export type HealthStatus = "ok" | "degraded";

export interface HealthResponse {
  status: HealthStatus;
  service: string;
  environment: string;
  api_prefix: string;
  database_configured: boolean;
  minio_configured: boolean;
  redis_configured: boolean;
  worker_contract: {
    backend_base_url: string;
    minio_public_url: string;
    worker_token_configured: boolean;
    lease_seconds: number;
  };
}

File: backend/app/schemas/health.py
from typing import Literal

from pydantic import BaseModel


class WorkerContractHealth(BaseModel):
    backend_base_url: str
    minio_public_url: str
    worker_token_configured: bool
    lease_seconds: int


class HealthResponse(BaseModel):
    status: Literal["ok", "degraded"]
    service: str
    environment: str
    api_prefix: str
    database_configured: bool
    minio_configured: bool
    redis_configured: bool
    worker_contract: WorkerContractHealth
